routes/spots.js

```diff
@@ -1,19 +1,44 @@
-const express = require('express');
-const { Spot, Review } = require('../db/models');
+const express = require("express");
+const { Spot, Review, Collection } = require("../db/models"); // Changed
+const { Op } = require("sequelize"); //Changed
 
 const router = express.Router();
 
-router.get('/', (req, res) => {
-    res.redirect('/');
+router.get("/", (req, res) => {
+  res.redirect("/");
 });
 
-router.get('/:id', async (req, res, next) => {
-    const id = req.params.id;
+router.get("/:id", async (req, res, next) => {
+  const id = req.params.id;
+  const userId = req.session.auth.userId;
 
-    const spot = await Spot.findByPk(id);
-    const reviews = await Review.findAll({where: { spotId: id}});
+  const spot = await Spot.findByPk(id);
+  const reviews = await Review.findAll({ where: { spotId: id } });
+  const customCollections = await Collection.findAll({

```

Comment:
Query to find all collections for the user, but then use javascript array methods to extract the two default collections to a separate array.

Reply:
Mitchell helped me come up with these 2 queries that should grab all the Reviews, Collections and SpotCollections tied to our current user:  spot variable to be used to render the spot info with associated reviews.  user variable to be used to perform client side validation checks to determine how to render our buttons on page load && avoid duplicate entries to our DB.  hopefully this helps, let me know if further clarification is needed. 

    const id = req.params.id;
    const userId = req.session.auth.userId;
    
    const spot = await Spot.findOne({where: {id}, include: Review });

    const user = await User.findOne({ where: { id: userId }, include: { model: Collection, as: 'collections', include: { model: Spot,   where: { id } } } } );


//use res.json(spot/user) and open page in browser to view expected output for each